numeric_ranges_algorithms/numeric_ranges_algorithms.md

```diff
@@ -0,0 +1,484 @@
+
+## Range categories and return types
+
+We make the parallel algorithms proposed here take sized random access ranges.  As a result, any parallel algorithms with an output range need to return an iterator to one past the last element of the input, in case there wasn't enough room in the output.  We make the non-parallel algorithms take (single-pass) sized `input_range` and `output_range`, for consistency with the C++17 parallel algorithms that only require _LegacyInputIterator_ and _LegacyOutputIterator_.  This suggests making the algorithms with output ranges -- `inclusive_scan` and `exclusive_scan` -- return both input and output iterators.  This is consistent both with the existing non-parallel ranges algorithms and with P3179R8 (e.g., `ranges::transform`).
+
+The `reduce` algorithm only takes an input range.  For the non-parallel overloads, this suggests that it should return both the computed value and the iterator to one past the last element of the input.  As we explain above, though, it would be more consistent with both the existing ranges algorithms and P3179R8 for `reduce` to return a single value.  We don't propose a separate algorithm `reduce_with_iter` here, though we would not oppose someone else doing so.
+
+P3179R8 does not aim for perfect consistency with the range categories accepted by existing ranges algorithms.  The algorithms proposed by P3179R8 differ in the following ways.
+
+1. P3179R8 uses a range, not an iterator, as the output parameter (see Section 2.7).
+
+2. P3179R8 requires that the ranges be sized (see Section 2.8).
+
+3. P3179R8 requires random access ranges (see Section 2.6).
+
+Of these differences, (1) and (2) could apply generally to all ranges algorithms, so we adopt them for this proposal.  This would make our proposal the first to add non-parallel range-as-output ranges algorithms to the Standard.  For arguments in favor of non-parallel algorithms taking a range as output, please refer to <a href="https://wg21.link/P3490R0">P3490R0</a>, "Justification for ranges as the output of parallel range algorithms."  (Despite the title, it has things to say about non-parallel algorithms too.)  Taking a range as output would prevent use of existing output-only iterators that do not have a separate sized sentinel type, like `std::back_insert_iterator`.  Such iterators would only work with the non-parallel algorithms.  P3490R0 shows that it is possible for both iterator-as-output and range-as-output overloads to coexist, so we follow P3179R8 by not proposing iterator-as-output algorithms here.
+
+Difference (1) relates to P3179R8 only proposing parallel algorithms.  It would make sense for us to relax this requirement for the non-parallel algorithms we propose.  This leaves us with two possibilities:

```

Comment:
Looks like you might have meant the difference (3) here, and not (1).

Reply:
Right, thank you!

Reply:
I've revised this bit and will update shortly.